#!/usr/bin/env python3
PACKAGE = "copilot_interface"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("thrusterToggle",   bool_t,   0, "Thrusters enabled safety param",  False)
gen.add("linearSense",    double_t,    0, "Linear ROV Sense", .5,  0, 1)
gen.add("angularSense",    double_t,    0, "Rotational ROV Sense", .5,  0, 1)
gen.add("verticalSense",    double_t,    0, "Vertical ROV Sense", .5,  0, 1)
gen.add("coralPhotos", bool_t, 0, "Toggle photogrammetry mode", False)
gen.add("pin5", bool_t, 0, "Toggle GPIO Board Pin 5", False)
gen.add("pump", bool_t, 0, "Toggle GPIO Board Pin 16", False)
gen.add("pin19", bool_t, 0, "Toggle GPIO Board Pin 19", False)
gen.add("pin23", bool_t, 0, "Toggle GPIO Board Pin 23", False)
gen.add("solenoid", bool_t, 0, "Toggle GPIO Board Pin 24", False)
gen.add("light", bool_t, 0, "Toggle GPIO Board Pin 25", False)

exit(gen.generate(PACKAGE, "copilotControls", "copilotControlParams"))
